src/ConversationEngine/Reasoners/ConversationalStateReasoner.php

```diff
@@ -23,12 +24,26 @@ class ConversationalStateReasoner
     /**
      * The user attribute should have the latest conversation record associated with it if this is
      * an existing user or no conversation record if it is a new user.
+     * If scenario id from request not matched with scenario id from user history record - returning new empty
+     * conversational state.
      * @param UserAttribute $user
+     * @param UtteranceAttribute|null $utterance
      * @return ConversationalState
      */
-    public static function determineConversationalStateForUser(UserAttribute $user): ConversationalState
-    {
+    public static function determineConversationalStateForUser(
+        UserAttribute      $user,
+        UtteranceAttribute $utterance = null
+    ): ConversationalState {
         if ($user->hasUserAttribute(UserAttribute::USER_HISTORY_RECORD)) {
+            if ($utterance) {
+                $previousScenarioId = $user->getUserHistoryRecord()->getScenarioId();
+                $currentScenarioId = $utterance->getScenarioId();
```

Comment:
As this method is called after we've established the user, and loaded the User Context, I think we'd be fine to replace this with something like the following. This means the new `$utterance` parameter could be removed.

```suggestion
                $currentScenarioId = ContextService::getAttributeValue('selected_scenario', UserContext::getComponentId());
```

Reply:
thanks, fixed